engine/crates/engine-v2/src/http_response/mod.rs

```diff
@@ -72,22 +61,16 @@ impl HttpGraphqlResponse {
         )
     }
 
-    pub(crate) fn build(
-        response: Response,
-        format: Option<StreamingFormat>,
-        metadata: HttpGraphqlResponseExtraMetadata,
-    ) -> Self {
-        let mut http_response = if let Some(format) = format {
+    pub(crate) fn build(response: Response, format: Option<StreamingFormat>) -> Self {
```

Comment:
:bug: **Bug**
The `build` function no longer initializes the `metadata` field, which was previously set to `HttpGraphqlResponseExtraMetadata::default()`. If any logic depends on the presence of this metadata, it could lead to unexpected behavior or errors.

```suggestion
// Ensure that any logic depending on metadata is updated to not rely on it, or handle it elsewhere if necessary.
```